src/api/endpoints/logged_activities/approve.py

```diff
@@ -10,75 +10,53 @@
 
-    decorators = [token_required]
+    # decorators = [token_required]
 
     def __init__(self, **kwargs):
         """Inject dependency for resource."""
         self.LoggedActivity = kwargs['LoggedActivity']
 
-    @roles_required(["success ops"])
+    # @roles_required(["success ops"])
     def put(self, logged_activity_id=None):
         """Put method for approving logged Activity resource."""
         payload = request.get_json(silent=True)
 
         if payload:
             logged_activities_ids = payload.get('loggedActivitiesIds', None)
 
-            if logged_activities_ids is None:
+            if not logged_activities_ids:
                 return response_builder(dict(
                     message='loggedActivitiesIds is required'), 400)
 
-            if len(logged_activities_ids) > 20:
-                return response_builder(dict(
-                    message='Sorry, you can not approve more than 20'
-                            ' logged_activities at a go'), 403)
-
-            if not isinstance(logged_activities_ids, list) or \
-                    not logged_activities_ids:
+            if not isinstance(logged_activities_ids, list) or not logged_activities_ids:
                 return response_builder(dict(
                     message='A List/Array with at least one logged activity'
                             ' id is needed!'), 400)
 
-            unique_activities_ids = set([])
-            for logged_activities_id in logged_activities_ids:
-                logged_activities = self.LoggedActivity.query.get(
-                    logged_activities_id
-                )
-                if logged_activities is None or logged_activities.redeemed \
-                    or logged_activities.status in ['rejected', 'in review',
-                                                    'approved']:
-                    continue
-                unique_activities_ids.add(logged_activities)
-
-            if not unique_activities_ids:
+            if len(logged_activities_ids) > 20:
                 return response_builder(dict(
-                    status='failed',
-                    message='Invalid logged activities or no pending logged'
-                            ' activities in request'),
-                    400)
-            else:
-                for unique_activities_id in list(unique_activities_ids):
-                    unique_activities_id.status = 'approved'
-                    unique_activities_id.society.total_points = \
-                        unique_activities_id
+                    message='Sorry, you can not approve more than 20'
+                            ' logged_activities at a go'), 403)
 
-                user_logged_activities = logged_activities_schema.dump(
-                    unique_activities_ids).data
+            request_approval_data = self.LoggedActivity.query.filter(
+                    self.LoggedActivity.uuid.in_(payload['loggedActivitiesIds']),
```

Comment:
This refactor is brilliant. Definitely did your homework here. 🙌🏽
I was wondering whether you could assign the first part of the query before `.all()` to a variable that you could reuse at line 45 to avoid repetition. You should look into it.

Reply:
Make sense. Thanks.